// expect/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.restoreSpies = exports.isSpy = exports.spyOn = exports.createSpy = exports.extend = exports.assert = undefined;

var _Expectation = require('./Expectation');

var _Expectation2 = _interopRequireDefault(_Expectation);

var _SpyUtils = require('./SpyUtils');

var _assert = require('./assert');

var _assert2 = _interopRequireDefault(_assert);

var _extend = require('./extend');

var _extend2 = _interopRequireDefault(_extend);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function expect(actual) {
  return new _Expectation2.default(actual);
}

// TODO: Maybe remove these eventually?
expect.createSpy = _SpyUtils.createSpy;
expect.spyOn = _SpyUtils.spyOn;
expect.isSpy = _SpyUtils.isSpy;
expect.restoreSpies = _SpyUtils.restoreSpies;
expect.assert = _assert2.default;
expect.extend = _extend2.default;

module.exports = expect;

exports.assert = _assert2.default;
exports.extend = _extend2.default;
exports.createSpy = _SpyUtils.createSpy;
exports.spyOn = _SpyUtils.spyOn;
exports.isSpy = _SpyUtils.isSpy;
exports.restoreSpies = _SpyUtils.restoreSpies;
